### ISSUE TITLE
a reusable framework that help user write his own indicator

### ISSUE THREAD

**Issue by GodRtio** [andredumas/techan.js#37]
Hi @andredumas,

I think it would be great if user can write his own indicator in javascript or coffee-script on the fly or save it to use later.

For example:
A simple fractal indicator:
if(data[i].high > data[i-1].high && data[i].high > data[i-2].high &&
   data[i].high > data[i+1].high && data[i].high > data[i+2].high){
   draw_symbol("fractal up" + i, i, fractalUpCode, color)
}
else if(data[i].low < data[i-1].low && data[i].low < data[i-2].low &&
   data[i].low < data[i+1].low && data[i].low < data[i+2].low ){
   draw_symbol("fractal down" + i, i, fractalDownCode, color)
}

Basically, we need a framework to draw something, like:
draw_line(id of line, position, type=trend|vertical|horizontal, style=solid|dash|dot, color)
draw_circle(...)
draw_rectangle(...)
draw_symbol(...)

And a way to manipulate data that have basic function like
access to any position of data to do calculation
min, max function
....

I don't know if it easy to write, but if you need any help, i can work with you.
Currently, i still read your code to understand the logic behind it.

Merry Christmas, @andredumas 

GodRtio
